pkg/k8s/client.go

```diff
@@ -155,6 +158,42 @@ func (c *Client) GetService(name, namespace string) *corev1.Service {
 	return nil
 }
 
+// ListSecrets returns a list of secrets
+func (c *Client) ListSecrets() []*corev1.Secret {
+	var secrets []*corev1.Secret
+
+	for _, secretPtr := range c.informers.List(osminformers.InformerKeySecret) {
+		if secretPtr == nil {
+			continue
+		}
+		secret, ok := secretPtr.(*corev1.Secret)
+		if !ok {
+			continue
+		}
+
+		secrets = append(secrets, secret)
+	}
+
+	return secrets
+}
+
+// GetSecret returns the secret for a given secret name and namespace
+func (c *Client) GetSecret(name, namespace string) *corev1.Secret {
+	secretIf, exists, err := c.informers.GetByKey(osminformers.InformerKeySecret, key(name, namespace))
+	if exists && err == nil {
+		secret := secretIf.(*corev1.Secret)
+		return secret
+	}
+	return nil
+}
+
+// UpdateSecretData updates the secret with the provided data
+func (c *Client) UpdateSecretData(ctx context.Context, secret *corev1.Secret, secretData map[string][]byte) error {
```

Comment:
can this just take in the secret object to update? ie: require the caller to do secret.Data = secretData.

Because if they've udpated anything else, then we're updating that as well, so that's a misnomer